interview-questions.md

```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
+  def change
+    add_column :sightings, :animal_id, :integer
+  end 
 
-  Researched answer:
+  Researched answer: rails generate migration and then add the changes to the migration file using the convention add_column :table, :name_of_column, :data_type
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: The RESTful routes that need params are: show, edit, update, and destroy. Create needs strong params to ensure only the desired data is stored to the database. These params are passed so the controller knows what id to call on to read/create the data. 
 
-  Researched answer:
+  Researched answer: The RESTful routes that need params are: show, edit, update and destroy. 
 
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: Three rails generator commands are: 
+  1. generate model - creates a basic table of keys and values that make up a database
+  2. generate migration - creates a migration to modify the shape of your table.
+  3. generate resource - creates a rails model that has all the routes connected to the controller and view built in.  
 
-  Researched answer:
+  Researched answer: generate model creates a migration, model and tests for the files that were generated.
+  generate migration implements transformations to your database.
+  generate resource creates a migration, model, tests, controller, views folder and resource routes.
 
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-method="GET"    /students          
+method="GET"    /students  index        
 
-method="GET"    /students/2  
+method="GET"    /students/2 show 
 
-method="GET"    /students/new
+method="GET"    /students/new new
 
-method="GET"    /students/edit/2    
+method="GET"    /students/edit/2 edit
 
-method="POST"   /students/       
+method="POST"   /students/ create
 
-method="PATCH"  /students/2      
+method="PATCH"  /students/2 update
 
-method="DELETE" /students/2      
+method="DELETE" /students/2 destroy 
 
 
 
 5. As a developer, you want to make an application that will help you manage your to do list. To get started, you want to create the API. Create 10 user stories that will help you get your application started.
+1 - Create a model named ToDoList and generate with resources to include the controller and routes.
+2 - Initialize the model with a task name and a description of the task
+3 - Migrate your db
+4 - Create controller method imdex that will access all items from the database
+5 - Create the show route in the controller that will look for the one item in the db
+6 - To be able to add new tasks, create the new route in the controller
+7 - Add a new item to the to do list by creating a RESTful route create 
+8 - Add 3 to do items to the list
+9 - Delete/destroy the first item on the list
+10 - Update/edit a task 
```

Comment:
Good work! User stories are generally from the perspective of what the user sees. This helps us as developers think about how we go about setting up our App and not just follow instructions of setting things up. This will get easier.  
// As a user I have access to a database called ToDoList that lets me make "to do lists" with items that can be marked as completed